### ISSUE TITLE
Selecting Numerator or Denominator

### ISSUE THREAD

**Issue by AdamCDunlap** [alex-ozdemir/algebra-dsl#47]
In `impl Indexable for Expression`, `get()` and `get_mut()` return `Expression`s, but if a user selects one half of a fraction, that half isn't an `Expression`. What do we want to do?

**Comment by alex-ozdemir** [comment]
Yeah, when I made the change to `Expr` (and changed the MathML output accordingly) one result was that the user could no longer select the top or bottom as a whole.

From Adam's email:

> It seems we have three options:
> 1) Don't support selection of a numerator or denominator, but still allow selection of each factor as siblings
> 2) Make division just be a single Expression for top and bottom
> 3) Don't assume that a node is an Expression
>
> I was never 100% convinced for the reason for removing Product since it doesn't seem much harder to match Division(Product(..), Product(..)), and maybe we should revisit that debate?

An interesting dilemma. The mathematical structure is better represented this way (and makes for easier transformations -- or at least simplified `simply_constants`) while the visual structure was better represented originally (and allows the user to select the top as an entity).

An important question: how hard is it to mock up structure A over a real structure B vs. B over A?

I can talk more about this Friday or later.

In the meantime, I would just use the current model -- if we switch we'll change the transforms.

**Comment by AdamCDunlap** [comment]
I'm not sure how to parse the phrase 

> mock up structure A over a real structure B vs. B over A
